src/placeos-driver/queue.cr

```diff
@@ -44,7 +45,7 @@ class PlaceOS::Driver::Queue
   # removes all jobs currently in the queue
   def clear(abort_current = false)
     old_queue = @queue
-    @queue = Priority::Queue(Task).new
+    @mutex.synchronize { @queue = Array(Task).new }
```

Comment:
I think the assignment should be done in the block here.

Also, why the new array instead of performing `Array#clear`?

Reply:
so we can queue new jobs as we abort the old